src/rgw/rgw_dmclock_sync_scheduler.cc

```diff
@@ -27,7 +27,7 @@ int SyncScheduler::add_request(const client_id& client, const ReqParams& params,
     }
     queue.request_completed();
     // Perform a blocking wait until the request callback is called
-    if (std::unique_lock<std::mutex> lk(req_mtx); rstate != ReqState::Wait) {
+    if (std::unique_lock<std::mutex> lk(req_mtx); rstate == ReqState::Wait) {
```

Comment:
```suggestion
    {
      std::unique_lock lock{req_mtx};
      req_cv.wait(lock, [&rstate] {return rstate != ReqState::Wait;});
    }
```

i'd suggest just replace the whole `if` block with the above one.

Reply:
I know what you mean，it will make the code more clear.
 but we need remove the below line:
req_cv.wait(lk, [&rstate] {return rstate != ReqState::Wait;});
    }  
if not ,will have build error

Reply:
> i'd suggest just replace the whole `if` block with the above one.

i hope you could make this more clearer.

Reply:
I have update the code, please review it, thank you.